src/zcl_abapgit_version.clas.abap

```diff
@@ -4,68 +4,132 @@ CLASS zcl_abapgit_version DEFINITION
+      RETURNING
+        VALUE(rv_result) TYPE i .
   PROTECTED SECTION.
   PRIVATE SECTION.
+
+    CLASS-METHODS version_to_numeric
+      IMPORTING
+        !iv_version       TYPE string
+      RETURNING
+        VALUE(rv_version) TYPE i.
 ENDCLASS.
 
 
 
 CLASS zcl_abapgit_version IMPLEMENTATION.
 
 
-  METHOD normalize.
+  METHOD check_dependant_version.
 
-    " Internal program version should be in format "XXX.XXX.XXX" or "vXXX.XXX.XXX"
-    CONSTANTS:
-      lc_version_pattern    TYPE string VALUE '^v?(\d{1,3}\.\d{1,3}\.\d{1,3})\s*$',
-      lc_prerelease_pattern TYPE string VALUE '^((rc|beta|alpha)\.\d{1,3})\s*$'.
+    CONSTANTS: lc_message TYPE string VALUE 'Current version is older than required'.
 
-    DATA: lv_version      TYPE string,
-          lv_prerelease   TYPE string,
-          lv_version_n    TYPE string,
-          lv_prerelease_n TYPE string.
+    IF is_dependant-major > is_current-major.
+      zcx_abapgit_exception=>raise( lc_message ).
+    ELSEIF is_dependant-major < is_current-major.
+      RETURN.
+    ENDIF.
 
-    SPLIT iv_version AT '-' INTO lv_version lv_prerelease.
+    IF is_dependant-minor > is_current-minor.
+      zcx_abapgit_exception=>raise( lc_message ).
+    ELSEIF is_dependant-minor < is_current-minor.
+      RETURN.
+    ENDIF.
 
-    FIND FIRST OCCURRENCE OF REGEX lc_version_pattern
-      IN lv_version SUBMATCHES lv_version_n.
+    IF is_dependant-patch > is_current-patch.
+      zcx_abapgit_exception=>raise( lc_message ).
+    ELSEIF is_dependant-patch < is_current-patch.
+      RETURN.
+    ENDIF.
 
-    IF lv_prerelease IS NOT INITIAL.
+    IF is_current-prerelase IS INITIAL.
+      RETURN.
+    ENDIF.
 
-      FIND FIRST OCCURRENCE OF REGEX lc_prerelease_pattern
-        IN lv_prerelease SUBMATCHES lv_prerelease_n.
+    CASE is_current-prerelase.
+      WHEN 'rc'.
+        IF is_dependant-prerelase = ''.
+          zcx_abapgit_exception=>raise( lc_message ).
+        ENDIF.
 
-    ENDIF.
+      WHEN 'beta'.
+        IF is_dependant-prerelase = '' OR is_dependant-prerelase = 'rc'.
+          zcx_abapgit_exception=>raise( lc_message ).
+        ENDIF.
 
-    IF lv_version_n IS INITIAL.
-      RETURN.
+      WHEN 'alpha'.
+        IF is_dependant-prerelase = '' OR is_dependant-prerelase = 'rc' OR is_dependant-prerelase = 'beta'.
+          zcx_abapgit_exception=>raise( lc_message ).
+        ENDIF.
+
+    ENDCASE.
+
+    IF is_dependant-prerelase = is_current-prerelase AND is_dependant-prerelase_patch > is_current-prerelase_patch.
+      zcx_abapgit_exception=>raise( lc_message ).
     ENDIF.
 
-    rv_version = lv_version_n.
+  ENDMETHOD.
 
-    IF lv_prerelease_n IS NOT INITIAL.
-      CONCATENATE rv_version '-' lv_prerelease_n INTO rv_version.
+
+  METHOD compare.
+
+    DATA: ls_version_a TYPE zif_abapgit_definitions=>ty_version,
+          ls_version_b TYPE zif_abapgit_definitions=>ty_version.
+
+    TRY.
+        IF is_a IS SUPPLIED.
```

Comment:
Imo `IS SUPPLIED` shouldn't be used anymore. If you wrap the call to the method in another one and pass an initial or no value there this will evaluate to true anyways.